tests/test_workspace.py

```diff
@@ -746,3 +746,14 @@ class FooWorkspace(pyhf.Workspace):
 
     combined = FooWorkspace.combine(ws, new_ws)
     assert isinstance(combined, FooWorkspace)
+
+
+def test_sorted(workspace_factory):
+    ws = workspace_factory()
+    # force the first sample in each channel to be last
+    for channel in ws['channels']:
+        channel['samples'][0]['name'] = 'zzzzlast'
+
+    new_ws = pyhf.Workspace.sorted(ws)
+    for channel in ws['channels']:
+        assert channel['samples'][-1]['name'] == 'zzzzlast'
```

Comment:
I'm a bit worried this is passing at the moment though, as doesn't this imply that `pyhf.Workspace.sorted(ws)` is modifying the `ws` passed to it along with the `new_ws` it returns? Do we want this side effect behavior and not something more functional?